exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -0,0 +1,245 @@
+   */
+  public static TestKit forService(Class<? extends ServiceModule> serviceModule) {
+    return new TestKit(singletonList(serviceModule), EmulatedNodeType.VALIDATOR, (short) 0, null);
+  }
+
+  /**
+   * Returns an instance of a service with the given service id and service class.
+   *
+   * @return the service instance or null if there is no service with such id
+   * @throws NullPointerException if the service with given id was not found
+   * @throws IllegalArgumentException if the service could not be cast to given class
+   */
+  public <T extends Service> T getService(short serviceId, Class<T> serviceClass) {
+    Service service = services.get(serviceId);
+    checkNotNull(service, "Service with given id=%s was not found", serviceId);
+    checkArgument(service.getClass().equals(serviceClass),
+        "Service (id=%s, class=%s) cannot be cast to %s",
+        serviceId, service.getClass().getCanonicalName(), serviceClass.getCanonicalName());
+    return serviceClass.cast(service);
+  }
+
+  /**
+   * Creates a user service adapter from the service module.
+   *
+   * @param moduleClass a class of the user service module
+   */
+  private UserServiceAdapter createUserServiceAdapter(Class<? extends ServiceModule> moduleClass) {
+    try {
+      Supplier<ServiceModule> moduleSupplier = new ReflectiveModuleSupplier(moduleClass);
+      ServiceModule serviceModule = moduleSupplier.get();
+      Injector serviceInjector = frameworkInjector.createChildInjector(serviceModule);
+      return serviceInjector.getInstance(UserServiceAdapter.class);
+    } catch (IllegalAccessException e) {
+      throw new IllegalArgumentException("Cannot access the no-arg module constructor", e);
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException("No no-arg constructor", e);
+    }
+  }
+
+  private native long nativeCreateTestKit(UserServiceAdapter[] services, boolean auditor,
+      short withValidatorCount, TimeProvider timeProvider);
+
+  private native long nativeCreateSnapshot(long nativeHandle);
+
+  private native Block nativeCreateBlock(long nativeHandle);
+
+  private native Block nativeCreateBlockWithTransactions(long nativeHandle, byte[][] transactions);
+
+  private native EmulatedNode nativeGetEmulatedNode(long nativeHandle);
+
+  /**
+   * Creates a new builder for the TestKit.
+   *
+   * @param nodeType type of the main TestKit node - either validator or auditor. Note that
+   * {@link Service#afterCommit(BlockCommittedEvent)} logic will only be called on the main TestKit node of this
+   * type
+   */
+  public static Builder builder(EmulatedNodeType nodeType) {
+    checkNotNull(nodeType);
+    return new Builder(nodeType);
+  }
+
+  /**
+   * Returns a list of user service adapters created from given service modules.
+   */
+  private List<UserServiceAdapter> toUserServiceAdapters(List<Class<? extends ServiceModule>> serviceModules) {
```

Comment:
These methods are used by the constructor only, therefore, I'd move them closer to it
 - ctor
   - toUserServiceAdapters
     - createUserServiceAdapter
   - populateServiceMap

Reply:
If they are private, have only a single usage and are adjacent to it, some documentation comments might no longer needed (unless they clarify something possibly unclear).

Reply:
… and `checkForDuplicateService` obviously